### PR TITLE
[REVIEW] Add notes to performance comparisons notebook

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
This PR adds a `note` section to the performance comparisons notebook to give users a disclaimer on what they need to do to run this notebook on lower-end hardware.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: docs/cudf/source/user_guide/performance_comparisons.ipynb [hunk 1]
```diff
@@ -8,13 +8,16 @@
...
   {
+   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
     "This notebook compares the performance of `cuDF` and `pandas`. The comparisons performed are on identical data sizes. This notebook primarily showcases the factor\n",
     "of speedups users can have when the similar `pandas` APIs are run on GPUs using `cudf`.\n",
     "\n",
-    "The hardware details used to run these performance comparisons are at the end of this page."
+    "The hardware details used to run these performance comparisons are at the end of this page.\n",
```
Comment (galipremsagar) [inline docs/cudf/source/user_guide/performance_comparisons.ipynb:18]: I've verified this claim holds true on a T4.

File: docs/cudf/source/user_guide/performance_comparisons/performance-comparisons.ipynb [hunk 2]
```diff
@@ -8,13 +8,16 @@
...
    "cell_type": "markdown",
    "metadata": {},
    "source": [
     "This notebook compares the performance of `cuDF` and `pandas`. The comparisons performed are on identical data sizes. This notebook primarily showcases the factor\n",
     "of speedups users can have when the similar `pandas` APIs are run on GPUs using `cudf`.\n",
     "\n",
-    "The hardware details used to run these performance comparisons are at the end of this page."
+    "The hardware details used to run these performance comparisons are at the end of this page.\n",
+    "\n",
+    "**Note**: This notebook is written to measure performance on modern NVIDIA hardware, for older NVIDIA hardware with lower GPU memory please consider lowering the `num_rows` values below by a factor of 2. Results may vary by data sizes, CPU & GPU used."
```
Comment (bdice) [inline docs/cudf/source/user_guide/performance_comparisons/performance-comparisons.ipynb:20]: I'm not sure that we should differentiate old/new here. We should focus on high-memory vs. low-memory. There are a range of relatively new consumer GPUs that would probably have too little memory for the largest data sizes here.
```suggestion
    "**Note**: This notebook is written to measure performance on NVIDIA GPUs with large memory. If running on hardware with lower memory, please consider lowering the `num_rows` values. Performance results may vary by data size, as well as the CPU and GPU used."
```

File: docs/cudf/source/user_guide/index.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ groupby
 guide-to-udfs
 cupy-interop
 options
-performance-comparisons
+performance_comparisons/index
```
Comment (bdice) [inline docs/cudf/source/user_guide/index.md:15]: Why did we name the folder `performance_comparisons` with an underscore, and the notebook `performance-comparisons.ipynb` with a dash?

**Review by bdice — APPROVED** [review]
LGTM, with one minor question on file naming.

**Review by raydouglass — APPROVED** [review]

**Review by vyasr — APPROVED** [review]
LGTM, thanks for the update

**Pull request merged**